modules/low-watermark/src/main/java/org/apache/ignite/internal/lowwatermark/LowWatermarkImpl.java

```diff
@@ -351,32 +315,25 @@ public void updateLowWatermark(HybridTimestamp newLowWatermark) {
 
     CompletableFuture<Void> updateAndNotify(HybridTimestamp newLowWatermark) {
         return inBusyLockAsync(busyLock, () -> {
-            // Wait until all the read-only transactions are finished before the new candidate, since no new RO transactions could be
-            // created, then we can safely promote the candidate as a new low watermark, store it in vault, and we can safely start cleaning
-            // up the stale/junk data in the tables.
-            return txManager.updateLowWatermark(newLowWatermark)
-                    .thenComposeAsync(unused -> inBusyLock(busyLock, () -> {
-                        vaultManager.put(LOW_WATERMARK_VAULT_KEY, ByteUtils.toBytes(newLowWatermark));
+            vaultManager.put(LOW_WATERMARK_VAULT_KEY, ByteUtils.toBytes(newLowWatermark));
 
-                        setLowWatermark(newLowWatermark);
+            setLowWatermark(newLowWatermark);
```

Comment:
To where do we set it?
Who's going to read it?
Shouldn't we do it _after_ notifications?

Reply:
I did not change the logic for setting the field.
This is how it was before.

I can make it so that we set the field only after the notifications, no problem, why do you think this is correct?
I think you can see for yourself who is reading this field using `LowWatermark#getLowWatermark` and `LowWatermark#getLowWatermarkSafe`.

Reply:
I don't know whether it's correct or not, there's no documentation that would explain to me - if I call `getLowWatermark`, would that be a candidate value of an actually processed value? If we're good with the first option, then that's fine. It would be nice to reflect that in Javadoc.
> I did not change the logic for setting the field.

_You did._
Previously, it has been done after `txManager.updateLowWatermark(newLowWatermark)`, now it's done before the corresponding action. I don't know what it affects, and why we didn't call `set*` before `update*` in the old version of the code. Something clearly has changed.

Reply:
fix it